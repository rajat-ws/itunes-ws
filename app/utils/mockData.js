export const MOCK_TRACK_DATA = {
  wrapperType: 'track',
  kind: 'song',
  artistId: 76531581,
  collectionId: 1068455603,
  trackId: 1068455608,
  artistName: 'Tom MacDonald',
  collectionName: 'See You Tomorrow',
  trackName: 'Greenday',
  collectionCensoredName: 'See You Tomorrow',
  trackCensoredName: 'Greenday',
  artistViewUrl: 'https://music.apple.com/us/artist/tom-macdonald/76531581?uo=4',
  collectionViewUrl: 'https://music.apple.com/us/album/greenday/1068455603?i=1068455608&uo=4',
  trackViewUrl: 'https://music.apple.com/us/album/greenday/1068455603?i=1068455608&uo=4',
  previewUrl:
    'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/51/bb/61/51bb6190-6865-3050-af01-0f64502bc376/mzaf_14313183610807291436.plus.aac.p.m4a',
  artworkUrl30:
    'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dd/64/78/dd6478bc-e684-1b69-2e4e-c7e1d6a36c0a/045635953260.jpg/30x30bb.jpg',
  artworkUrl60:
    'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dd/64/78/dd6478bc-e684-1b69-2e4e-c7e1d6a36c0a/045635953260.jpg/60x60bb.jpg',
  artworkUrl100:
    'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dd/64/78/dd6478bc-e684-1b69-2e4e-c7e1d6a36c0a/045635953260.jpg/100x100bb.jpg',
  collectionPrice: 9.99,
  trackPrice: 0.99,
  releaseDate: '2015-12-21T12:00:00Z',
  collectionExplicitness: 'explicit',
  trackExplicitness: 'explicit',
  discCount: 1,
  discNumber: 1,
  trackCount: 17,
  trackNumber: 3,
  trackTimeMillis: 256520,
  country: 'USA',
  currency: 'USD',
  primaryGenreName: 'Hip-Hop/Rap',
  contentAdvisoryRating: 'Explicit',
  isStreamable: true
};
