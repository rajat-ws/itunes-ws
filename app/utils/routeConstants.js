export default {
  itunes: {
    route: '/',
    props: {
      maxWidth: 32,
      padding: 20
    },
    exact: true
  },
  trackDetails: {
    route: '/tracks/:trackId',
    exact: true
  }
};
