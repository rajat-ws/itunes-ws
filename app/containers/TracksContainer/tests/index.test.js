/*
 ** Tests for TracksContainer
 */
import React from 'react';
import { renderProvider, renderWithIntl, timeout } from '@app/utils/testUtils';
import { fireEvent } from '@testing-library/dom';
import { TracksContainerTest as TracksContainer } from '../index';

describe('TracksContainer Tests', () => {
  //testing for dispatchRequestTracksData
  let mockdispatchRequestTracksData;

  beforeEach(() => {
    mockdispatchRequestTracksData = jest.fn();
  });
  it('should render and match to the snapshot', () => {
    const { baseElement } = renderWithIntl(
      <TracksContainer
        dispatchRequestTracksData={mockdispatchRequestTracksData}
        tracksData={{ 1: { name: 'song1' }, 2: { name: 'song1' } }}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  // dispatch track names
  it('should call dispatchRequestTracksData on change', async () => {
    const { queryByTestId } = renderWithIntl(
      <TracksContainer
        dispatchRequestTracksData={mockdispatchRequestTracksData}
        tracksData={{ 1: { name: 'song1' }, 2: { name: 'song1' } }}
      />
    );
    fireEvent.change(queryByTestId('search-bar'), {
      target: { value: 'Arijit' }
    });
    await timeout(500);
    expect(mockdispatchRequestTracksData).toBeCalled();
  });

  // testing For component
  it('should render For component when tracksData is available', async () => {
    let tracksDummyData = { 1: { name: 'song1' }, 2: { name: 'song1' } };
    await timeout(500);
    const { queryByTestId } = renderProvider(<TracksContainer tracksData={tracksDummyData} />);
    expect(queryByTestId('for')).toBeInTheDocument();
  });

  // testing for dispatchClearTracksData
  it('should trigger the disptachsingleTrackData while search input is being emptied', async () => {
    let searchTrackNamesSpy = jest.fn();
    let clearTracksDataSpy = jest.fn();
    const { queryByTestId } = renderWithIntl(
      <TracksContainer
        tracksData={{ 1: { name: 'song1' }, 2: { name: 'song1' } }}
        dispatchRequestTracksData={searchTrackNamesSpy}
        dispatchClearTracksData={clearTracksDataSpy}
      />
    );

    fireEvent.change(queryByTestId('search-bar'), {
      target: { value: 'a' }
    });
    await timeout(500);
    expect(searchTrackNamesSpy).toBeCalled();
    fireEvent.change(queryByTestId('search-bar'), {
      target: { value: '' }
    });
    await timeout(500);
    expect(clearTracksDataSpy).toBeCalled();
  });
});
