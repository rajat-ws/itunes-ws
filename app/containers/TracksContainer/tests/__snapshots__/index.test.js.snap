// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracksContainer Tests should render and match to the snapshot 1`] = `
<body>
  <div>
    <div
      class="TracksContainer__Container-jv1ui9-2 buqqOa"
      data-testid="tracks-container"
    >
      <div
        class="TracksContainer__StyledSearchContainer-jv1ui9-1 csDMaK"
      >
        <div
          class="ant-card ant-card-bordered TracksContainer__TitleCard-jv1ui9-0 byhIsA"
          maxwidth="32"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                Itunes store
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <p
              class="T__StyledText-gjlic1-0 fIbhJF"
              data-testid="t"
            >
              Search for tracks/artists
            </p>
            <span
              class="ant-input-group-wrapper ant-input-search"
            >
              <span
                class="ant-input-wrapper ant-input-group"
              >
                <input
                  class="ant-input"
                  data-testid="search-bar"
                  type="text"
                  value=""
                />
                <span
                  class="ant-input-group-addon"
                >
                  <button
                    class="ant-btn ant-btn-icon-only ant-input-search-button"
                    type="button"
                  >
                    <span
                      aria-label="search"
                      class="anticon anticon-search"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="search"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card ant-card-bordered TracksContainer__TitleCard-jv1ui9-0 fTXtPF"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="TracksContainer__TrackGrid-jv1ui9-3 gvaiGt"
          data-testid="for"
          orientation="row"
        >
          <div
            class="ant-card ant-card-bordered TrackComponent__TrackCardContainer-sc-1y7fh5t-0 jBWOdU"
            data-testid="track-component"
          >
            <div
              class="ant-card-body"
            >
              <div
                class="ant-image"
              >
                <img
                  class="ant-image-img"
                />
                <div
                  class="ant-image-mask"
                >
                  <div
                    class="ant-image-mask-info"
                  >
                    <span
                      aria-label="eye"
                      class="anticon anticon-eye"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="eye"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                        />
                      </svg>
                    </span>
                    Preview
                  </div>
                </div>
              </div>
              <div
                class="TrackComponent__StyledDescription-sc-1y7fh5t-1 ehlPZI"
              >
                <div
                  class="ant-typography"
                >
                  No artist name available
                </div>
                <div
                  class="ant-typography"
                >
                  No collection name available
                </div>
                <div
                  class="ant-typography"
                >
                  No track name available
                </div>
              </div>
              <div
                class="TrackComponent__ButtonWrapper-sc-1y7fh5t-5 dehKAG"
              >
                <button
                  class="TrackComponent__ShowDetailsBtn-sc-1y7fh5t-4 cxivdz"
                >
                  <span
                    class="TrackComponent__ButtonLabel-sc-1y7fh5t-6 ftUulf"
                  >
                     Show Details 
                  </span>
                </button>
                <button
                  class="TrackComponent__PlayTrackBtn-sc-1y7fh5t-3 khRFCr"
                >
                  <span
                    class="TrackComponent__ButtonLabel-sc-1y7fh5t-6 ftUulf"
                  >
                     Play 
                  </span>
                </button>
              </div>
              <audio
                data-testid="trackAudio"
              />
            </div>
          </div>
          <div
            class="ant-card ant-card-bordered TrackComponent__TrackCardContainer-sc-1y7fh5t-0 jBWOdU"
            data-testid="track-component"
          >
            <div
              class="ant-card-body"
            >
              <div
                class="ant-image"
              >
                <img
                  class="ant-image-img"
                />
                <div
                  class="ant-image-mask"
                >
                  <div
                    class="ant-image-mask-info"
                  >
                    <span
                      aria-label="eye"
                      class="anticon anticon-eye"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="eye"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                        />
                      </svg>
                    </span>
                    Preview
                  </div>
                </div>
              </div>
              <div
                class="TrackComponent__StyledDescription-sc-1y7fh5t-1 ehlPZI"
              >
                <div
                  class="ant-typography"
                >
                  No artist name available
                </div>
                <div
                  class="ant-typography"
                >
                  No collection name available
                </div>
                <div
                  class="ant-typography"
                >
                  No track name available
                </div>
              </div>
              <div
                class="TrackComponent__ButtonWrapper-sc-1y7fh5t-5 dehKAG"
              >
                <button
                  class="TrackComponent__ShowDetailsBtn-sc-1y7fh5t-4 cxivdz"
                >
                  <span
                    class="TrackComponent__ButtonLabel-sc-1y7fh5t-6 ftUulf"
                  >
                     Show Details 
                  </span>
                </button>
                <button
                  class="TrackComponent__PlayTrackBtn-sc-1y7fh5t-3 khRFCr"
                >
                  <span
                    class="TrackComponent__ButtonLabel-sc-1y7fh5t-6 ftUulf"
                  >
                     Play 
                  </span>
                </button>
              </div>
              <audio
                data-testid="trackAudio"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
